### PR TITLE
Use logging in modelica system syntron

### PR DESCRIPTION
**Pull request by syntron**
Update ModelicaSystem to use Exceptions / the python logging interface (see #227)

It is based on #226 due to the develop history; if needed it could be rebased or parts could be extracted

### CODE REVIEW

**Comment by adeas31** [comment]
Please rebase this on master so the external dll path commits are not included.

**Comment by syntron** [comment]
Rebased to master (as #226 is included now)

File: OMPython/__init__.py [hunk 1]
```diff
@@ -906,29 +916,33 @@ def loadLibrary(self, verbose):
...
-                    print("| info | loadLibrary() failed, Unknown type detected: ", element , " is of type ",  type(element), ", The following patterns are supported\n1)[\"Modelica\"]\n2)[(\"Modelica\",\"3.2.3\"), \"PowerSystems\"]\n")
+                    raise ModelicaSystemError("loadLibrary() failed, Unknown type detected: " +
+                                              "{} is of type {}, ".format(element, type(element)) +
+                                              "The following patterns are supported:\n" +
+                                              "1)[\"Modelica\"]\n" +
+                                              "2)[(\"Modelica\",\"3.2.3\"), \"PowerSystems\"]\n")
                 ## Show notification or warnings to the user when verbose=True OR if some error occurred i.e., not result
-                if verbose or not result:
-                    print(self.requestApi('getErrorString'))
+                if not result:
```
Comment (adeas31) [inline OMPython/__init__.py:925]: Check for `self._verbose` flag here. `loadLibrary` raises some warnings and notifications that are good to know specially when `verbose` is set.

File: OMPython/__init__.py [hunk 2]
```diff
@@ -1446,23 +1475,27 @@ def setContinuous(self, cvals):  # 13
...
+        return self.setMethodHelper(pvals, self.paramlist, "parameter", self.overridevariables)
 
-    def isParameterChangeable(self, name, value, verbose):
+    def isParameterChangeable(self, name, value):
         q = self.getQuantities(name)
         if (q[0]["changeable"] == "false"):
-            if verbose:
-                print("| info |  setParameters() failed : It is not possible to set the following signal " + "\"" + name + "\"" + ", It seems to be structural, final, protected or evaluated or has a non-constant binding, use sendExpression(setParameterValue("+ self.modelName + ", " + name + ", " + value + "), parsed=false)" + " and rebuild the model using buildModel() API")
+            if self._verbose:
+                logger.warning("setParameters() failed : It is not possible to set " +
```
Comment (adeas31) [inline OMPython/__init__.py:1494]: Should be `logger.info`

File: OMPython/__init__.py [hunk 3]
```diff
@@ -967,41 +981,49 @@ def _run_cmd(self, cmd: list, verbose: bool = True):
...
+        if errstr is None or not errstr:
+            return
+
+        self._raise_error(errstr=errstr)
+
+    def _raise_error(self, errstr: str):
+        if self._raiseerrors:
+            raise ModelicaSystemError("OM error: {}".format(errstr))
+        else:
+            logger.warning(errstr)
```
Comment (adeas31) [inline OMPython/__init__.py:1005]: This should be `logger.error`

**Review by adeas31 — CHANGES_REQUESTED** [review]

**Comment by syntron** [comment]
@adeas31 update based on your comments

File: OMPython/__init__.py [hunk 4]
```diff
@@ -860,36 +870,36 @@ def __init__(self, fileName=None, modelName=None, lmodel=[], useCorba=False, com
...
-    def loadFile(self, verbose):
+    def loadFile(self):
         # load file
-        loadFileExp="".join(["loadFile(","\"",self.fileName,"\"",")"]).replace("\\","/")
+        loadFileExp = "".join(["loadFile(", "\"", self.fileName, "\"", ")"]).replace("\\", "/")
         loadMsg = self.getconn.sendExpression(loadFileExp)
         ## Show notification or warnings to the user when verbose=True OR if some error occurred i.e., not result
-        if verbose or not loadMsg:
-            return print(self.getconn.sendExpression("getErrorString()"))
+        if not loadMsg:
```
Comment (adeas31) [inline OMPython/__init__.py:898]: Forgot to add check for `self._verbose` flag.

**Review by adeas31 — APPROVED** [review]
LGTM

**Comment by adeas31** [comment]
@arun3688 can you please review and test it.

File: OMPython/__init__.py (hunk 4 above)
Comment (syntron) [reply OMPython/__init__.py]: Done in an added commit; see below ...

**Pull request merged**